File: 백준/Gold/5430. AC/AC.py
from collections import deque

t = int(input())
for i in range(t):
    p = input().strip()
    n = int(input())
    arr = input().strip()
    q = deque(arr[1:-1].split(','))
    flag = 1
    if n == 0:
        q = deque()

    R = 0
    for a in p:
        if a == 'R':
            R += 1
        elif a == 'D':
            if len(q) == 0:
                print("error")
                flag = 0
                break
            else:
                if R % 2 == 0:
                    q.popleft()
                else:
                    q.pop()
    if flag == 0:
        continue
    else:
        if R % 2 == 0:
            print('[' + ",".join(q) + ']')
        else:
            q.reverse()
            print('[' + ",".join(q) + ']')

File: 백준/Gold/5430. AC/AC.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "ex.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

const t = Number(input[0]);
for (let i = 1; i <= t * 3; i += 3) {
  const fn = input[i].split("");
  const n = Number(input[i + 1]);
  const arr = JSON.parse(input[i + 2]);
  let isReverse = false;
  let isError = false;
  for (const f of fn) {
    if (f === "R") {
      isReverse = !isReverse;
    }
    if (f === "D") {
      if (arr.length === 0) {
        isError = true;
      }
      if (isReverse) {
        arr.pop();
      }
      if (!isReverse) {
        arr.shift();
      }
    }
  }
  if (isError) {
    console.log("error");
  } else {
    if (isReverse) {
      console.log(JSON.stringify(arr.reverse()));
    } else {
      console.log(JSON.stringify(arr));
    }
  }
}
